chemprop/cli/train.py

```diff
@@ -650,6 +641,63 @@ def build_splits(args, format_kwargs, featurization_kwargs):
     return train_data, val_data, test_data
 
 
+def summarize(args, dataset: _MolGraphDatasetMixin) -> tuple[list, list]:
+    if args.task_type == "spectral":
+        pass
+    elif args.task_type in ["regression", "regression-mve", "regression-evidential"]:
+        y = dataset.Y
+        y_mean = y.mean()
+        y_std = y.std()
+        y_median = np.median(y)
+        mean_dev_abs = np.abs(y - y_mean)
+        frac_1_sigma = (mean_dev_abs < y_std).sum() / len(y)
+        frac_2_sigma = (mean_dev_abs < 2 * y_std).sum() / len(y)
+        column_headers = ["Statistic", "Value"]
+        table_rows = [
+            ["Num. datapoints", f"{len(y)}"],
+            ["Mean", f"{y_mean:0.2f}"],
+            ["Std. dev.", f"{y_std:0.2f}"],
+            ["Median", f"{y_median:0.2f}"],
+            ["% within 1 s.d.", f"{frac_1_sigma:0.0%}"],
+            ["% within 2 s.d.", f"{frac_2_sigma:0.0%}"],
+        ]
+        return (column_headers, table_rows)
+    elif args.task_type in [
+        "classification",
+        "classification-dirichlet",
+        "multiclass",
+        "multiclass-dirichlet",
+    ]:
+        y = dataset.Y
+        mask = np.isnan(y)
+        classes = np.sort(np.unique(y[~mask]))
+        class_counts = (classes[:, None] == y[None, :]).sum(1)
+        class_fracs = class_counts / len(y)
+        nan_count = (mask).sum()
+        nan_frac = nan_count / len(y)
+        column_headers = ["Class", "Count", "Percent"]
+        table_rows = [
+            (f"{k}", f"{class_counts[i]}", f"{class_fracs[i]:0.0%}") for i, k in enumerate(classes)
+        ]
+        table_rows.append(["Nan", f"{nan_count}", f"{nan_frac:0.0%}"])
+        table_rows.append(["Total", f"{len(y)}", f"{100.00}%"])
+        return (column_headers, table_rows)
+    else:
+        raise ValueError(f"Task type {args.task_type} was not recognized.")
+
+
+def build_table(column_headers: list[str], table_rows: list[str], title: str | None = None) -> str:
+    right_justified_columns = [
+        Column(header=column_header, justify="right") for column_header in column_headers
+    ]
+    table = Table(*right_justified_columns, title=title)
+    for row in table_rows:
+        table.add_row(*row)
+    console = Console(record=True)
+    console.print(table)
+    return console.export_text()
```

Comment:
@KnathanM @donerancl I believe this is the source of the double-printing when `args.log is None`. The `Console` object is an object-oriented `stdout`, so `console.print()` will first print to `stdout`. Later on, our logs also print to `stdout` (because they're pointed there), thus printing the table again. I think we need:

```suggestion
    console = Console(record=True)
    with console.capture() as capture:
        console.print(table)
    
    return capture.get()
```

Reply:
@davidegraff This seems to have fixed the double-printing, but some characters don't appear correctly in the log file... The column titles look like this 
```┃[1m [0m[1mClass[0m[1m [0m┃[1m [0m[1mCount[0m[1m [0m┃[1m [0m[1mPercent[0m[1m [0m┃```
I'm trying to find something online that will fix it...

Reply:
I guess we could also just point the `Console` to `/etc/null` and keep this the same. Does that solve the issue?

Reply:
@KnathanM and I found a solution for this, it is working now